src/Protocols/Avalon/MemMap/AvalonMemMap.hs

```diff
@@ -0,0 +1,1245 @@
+{-|
+Types and instance declarations for the Avalon memory mapped protocol
+(http://www1.cs.columbia.edu/~sedwards/classes/2009/4840/mnl_avalon_spec.pdf).
+Non-required fields can be easily toggled by the user. The @data@ and
+@outputenable@ fields are not supported since we would need bidirectional data
+ports. The @resetrequest@ field is also not supported since this does not get
+transferred around, but rather gets send "outwards" to whoever is controlling
+the reset signal of the circuit.
+-}
+
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Protocols.Avalon.MemMap.AvalonMemMap
+  ( -- * Configuration types
+    AvalonMMSharedConfig(..)
+  , AvalonMMSubordinateConfig(..)
+  , AvalonMMManagerConfig(..)
```

Comment:
```suggestion
  , AvalonMmManagerConfig(..)
```

I think this makes it a little bit easier to read. We had a discussion on it: most people preferred `Xx` over `XX` (Rust-like casing).